solidity/ecdsa/test/WalletRegistry.Wallets.test.ts

```diff
@@ -20,6 +23,183 @@ describe("WalletRegistry - Wallets", async () => {
+            expectedWalletID:
+              "0xa8b7226c57b544536f7bf805ef75c7b831488398da117644839f650c5be6cbe0",
+          },
+        ]
+        testData.forEach((test) => {
+          let walletID: string
+          let dkgResult: DkgResult
+
+          before("create a wallet", async () => {
+            await createSnapshot()
+            ;({ walletID, dkgResult } = await createNewWallet(
+              walletRegistry,
+              walletOwner.wallet,
+              test.publicKey
+            ))
+          })
+
+          after(async () => {
+            await restoreSnapshot()
+          })
+
+          it("should register wallet's details", async () => {
+            const wallet = await walletRegistry.getWallet(walletID)
+
+            expect(
+              wallet.membersIdsHash,
+              "unexpected members ids hash"
+            ).to.be.equal(hashUint32Array(dkgResult.members))
+
+            expect(wallet.publicKeyX, "unexpected public key X").to.be.equal(
+              ethers.utils.hexDataSlice(test.publicKey, 0, 32)
+            )
+            expect(wallet.publicKeyY, "unexpected public key Y").to.be.equal(
+              ethers.utils.hexDataSlice(test.publicKey, 32)
+            )
+          })
+
+          describe("getWalletPublicKey", async () => {
+            it("should return concatenated public key", async () => {
+              expect(
+                await walletRegistry.getWalletPublicKey(walletID),
+                "unexpected concatenated public key"
+              ).to.be.equal(test.publicKey)
+            })
+          })
+
+          it("should calculate wallet id", async () => {
+            expect(walletID, "unexpected walletID").to.be.equal(
+              test.expectedWalletID
+            )
+          })
+        })
+      })
+
+      context("with invalid public key", async () => {
+        const testData = [
+          {
+            context: "with too short public key",
+            publicKey: ethers.utils.randomBytes(63),
+            expectedError: "Invalid length of the public key",
+          },
+          {
+            context: "with too long public key",
+            publicKey: ethers.utils.randomBytes(65),
+            expectedError: "Invalid length of the public key",
+          },
+        ]
+
+        testData.forEach((test) => {
+          context(test.context, async () => {
+            before("create a wallet", async () => {
+              await createSnapshot()
+            })
+
+            after(async () => {
+              await restoreSnapshot()
+            })
+
+            it("should revert", async () => {
+              await expect(
+                createNewWallet(
+                  walletRegistry,
+                  walletOwner.wallet,
+                  test.publicKey
+                )
+              ).to.be.revertedWith(test.expectedError)
+            })
+          })
+        })
+      })
+    })
+
+    context("with wallet registered", async () => {
```

Comment:
Non-blocking but worth keeping in mind: we do not need `async` for context fn in the majority of cases.

Reply:
If it doesn't cause any problems I like that we keep it consistent and don't try to figure out where it should be placed and where it's not needed. IMHO it's more important for `await` with expects where it's not always needed but if we miss it in a place where it should be added then we will end up with a false-positive test result.

Reply:
I might be missing something, can you give me one example in our codebase of where it is really needed?

Reply:
Do you mean an example of `await` with `expect`? Here we have it, but if it wasn't provided we would not catch a test failure: https://github.com/keep-network/keep-core/blob/b6768d9bd326f4fe67bb189fafd27b5aff641c0c/solidity/ecdsa/test/WalletRegistry.Wallets.test.ts#L173-L177

I've seen numerous of times situations where we were having `expect` without an `await` where it was really necessary. This is why I prefer to add `await` to all the `expect`s, even if it's redundant.